app/lib/omni_auth/strategies/dev_auth.rb

```diff
@@ -0,0 +1,67 @@
+module OmniAuth
+  module Strategies
+    class DevAuth
+      # IMPORTANT NOTE: This OmniAuth strategy is intended for local
+      # development purposes only.
+      #
+      # When a user clicks the "Sign in" button with this strategy,
+      # they are directed to the DevAuth form where they can select
+      # the email of an existing user to log in as. To add an admin
+      # user, run "bundle exec rails db:seed" if it has not been
+      # done already.
+      #
+      # The list of user emails includes a Non Authorised user's
+      # email (NO_AUTH_EMAIL), which developers can use to simulate
+      # a non-authorised authenticated user.
+      #
+      # During the callback phase, this strategy searches for a user
+      # with the given email address in the local database. If a user
+      # with a matching email address is found and is authenticated,
+      # the strategy constructs an auth hash based on the user's
+      # information in the database.
+      #
+      # In cases where the strategy is unable to find an authenticated
```

Comment:
Wondering if this could be more confusing - why have it guess? My thinking is that if we have the dev setup steps in the README for local dev then people will just do that. If they haven't, it doesn't work, which is a good thing. Happy either way though really, I can see this being more convenient in some ways.